server/types/name.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"reflect"
+	"unicode/utf8"
+
+	"github.com/lib/pq/oid"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+	"github.com/dolthub/vitess/go/sqltypes"
+	"github.com/dolthub/vitess/go/vt/proto/query"
+)
+
+const NameLength = 63
+
+// Name is a 63-byte internal type for object names.
+var Name = NameType{Length: NameLength}
+
+// NameType is the extended type implementation of the PostgreSQL name.
+type NameType struct {
+	// Length represents the maximum number of characters that the type may hold.
+	Length uint32
+}
+
+var _ DoltgresType = NameType{}
+
+// BaseID implements the DoltgresType interface.
+func (b NameType) BaseID() DoltgresTypeBaseID {
+	return DoltgresTypeBaseID_Name
+}
+
+// CollationCoercibility implements the DoltgresType interface.
+func (b NameType) CollationCoercibility(ctx *sql.Context) (collation sql.CollationID, coercibility byte) {
+	return sql.Collation_binary, 5
+}
+
+// Compare implements the DoltgresType interface.
+func (b NameType) Compare(v1 any, v2 any) (int, error) {
+	if v1 == nil && v2 == nil {
+		return 0, nil
+	} else if v1 != nil && v2 == nil {
+		return 1, nil
+	} else if v1 == nil && v2 != nil {
+		return -1, nil
+	}
+
+	ac, _, err := b.Convert(v1)
+	if err != nil {
+		return 0, err
+	}
+	bc, _, err := b.Convert(v2)
+	if err != nil {
+		return 0, err
+	}
+
+	ab := ac.(string)
+	bb := bc.(string)
+	if ab == bb {
+		return 0, nil
+	} else if ab < bb {
+		return -1, nil
+	} else {
+		return 1, nil
+	}
+}
+
+// Convert implements the DoltgresType interface.
+func (b NameType) Convert(val any) (any, sql.ConvertInRange, error) {
+	// TODO: need to check if this always truncates for values that are too large, or if it's just the default behavior
```

Comment:
Don't need to do anything for this since we'll be removing `Convert` anyway, but in this case where the logic is non-trivial and exactly the same as another (`VarChar` in this case), we should factor it out to another function, or you can call it directly with `return (VarCharType{Length: b.Length}).Convert(val)` or something similar.